sdk/core/azure-core/tests/perf_tests/upload_binary.py

```diff
@@ -0,0 +1,94 @@
+from devtools_testutils.perfstress_tests import RandomStream, AsyncRandomStream
+
+from azure.core.rest import HttpRequest
+from azure.core.exceptions import (
+    HttpResponseError,
+    map_error,
+)
+from azure.core.pipeline.transport import (
+    RequestsTransport,
+    AsyncioRequestsTransport,
+)
+from ._test_base import _BlobTest
+
+import logging
+import sys
+handler = logging.StreamHandler(stream=sys.stdout)
+logger = logging.getLogger('azure')
+logger.setLevel(logging.DEBUG)
+logger.addHandler(handler)
+
+class UploadBinaryDataTest(_BlobTest):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        blob_name = "uploadtest"
+        self.blob_endpoint = f"{self.account_endpoint}{self.container_name}/{blob_name}"
+        self.upload_stream = RandomStream(self.args.size)
+        self.upload_stream_async = AsyncRandomStream(self.args.size)
+        # TODO: investigate size-1 issue for requests transports - https://github.com/Azure/azure-sdk-for-python/issues/33997
+        if self.sync_transport == RequestsTransport:
+            self.sync_size = self.args.size - 1
+        else:
+            self.sync_size = self.args.size
+        if self.async_transport == AsyncioRequestsTransport:
+            self.async_size = self.args.size - 1
+        else:
+            self.async_size = self.args.size
+
+    # TODO: If seeing "the request verb is invalid" error frequently: https://github.com/Azure/azure-sdk-for-python/issues/33999
+    def run_sync(self):
+        self.upload_stream.reset()
+        current_time = format_date_time(time())
+        request = HttpRequest(
+            method="PUT",
+            url=self.blob_endpoint,
+            params={},
+            headers={
+                'x-ms-date': current_time,
+                'x-ms-blob-type': 'BlockBlob',
+                'Content-Length': str(self.sync_size),
+                'x-ms-version': self.api_version,
+                'Content-Type': 'application/octet-stream',
+            },
+            content=self.upload_stream
+        )
+        response = self.pipeline_client._pipeline.run(
+            request
+        ).http_response
+        if response.status_code not in [201]:
+            map_error(status_code=response.status_code, response=response, error_map=self.error_map)
+            raise HttpResponseError(response=response)
+
+    async def run_async(self):
+        self.upload_stream_async.reset()
+        current_time = format_date_time(time())
+        request = HttpRequest(
+            method="PUT",
+            url=self.blob_endpoint,
+            params={},
+            headers={
+                'x-ms-date': current_time,
+                'x-ms-blob-type': 'BlockBlob',
+                'Content-Length': str(self.async_size),
+                'x-ms-version': self.api_version,
+                'Content-Type': 'application/octet-stream',
+            },
+            content=self.upload_stream_async
```

Comment:
Could we add a TODO comment for making sure the sync stream at least works as well?

Reply:
resolved: This was not working b/c aiohttp creates a BytesIOPayload by checking that the type of data is `BytesIO` [[here](https://github.com/aio-libs/aiohttp/blob/2349cba245c0015c488f89d529282432454ad65b/aiohttp/payload.py#L116)]. However, RandomStream was not inheriting from BytesIO while AsyncRandomStream was. Having RandomStream inherit from BytesIO resolved the issue.